internal/cluster/node.go

```diff
@@ -91,15 +102,20 @@ func (n *NodeInfo) populateNodeIpList() error {
 }
 
 func GetNodeIP(client client.Client) (string, error) {
-	nodeInfo := &NodeInfo{
-		Client: client,
-	}
-	nodeIPs, err := nodeInfo.getNodeExternalIpList(client)
-	if err != nil {
+	nodeInfo.Client = client
+	nodeIPs, err := nodeInfo.getNodeExternalIpList()
+	if err != nil || len(nodeIPs) == 0 {
 		log.Error(err, "Getting NodeIP From kube-api-server")
 		os.Exit(1)
 	}
+
 	nodeIP := nodeIPs[0]
 	log.Info("nodeIP selected", "nodeIP ", nodeIP)
 	return nodeIP, err
 }
+
+func GetNodeExternalIpList() []string {
+	nodeInfo.Lock()
+	defer nodeInfo.Unlock()
+	return nodeInfo.ExternalIP
```

Comment:
`nodeInfo.ExternalIP` should be `nodeInfo.ExternalIPList` or `nodeInfo.ExternalIPs` since it is a list of Ip addresses

Reply:
yeah